packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -268,7 +268,7 @@ class _DevFSHttpWriter {
       request.headers.removeAll(HttpHeaders.ACCEPT_ENCODING);
       request.headers.add('dev_fs_name', fsName);
       request.headers.add('dev_fs_path_b64',
-                          BASE64.encode(UTF8.encode(devicePath)));
+                          BASE64.encode(UTF8.encode(devicePath.path)));
```

Comment:
I added a TODO to transfer the full Uri when the remote end supports it. I think the remote end should be switched to use `Uri.parse` instead of `Uri.file` and we should always be communicating in platform-independent URIs to make things easier.